packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,637 @@
+import {
+  Diagnostic,
+  DiagnosticCollector,
+  NoTarget,
+  SourceFile,
+  createDiagnosticCollector,
+  resolvePath,
+} from "@typespec/compiler";
+import { HttpStatusCodeRange } from "@typespec/http";
+import { resolveOperationId } from "@typespec/openapi";
+import {
+  SdkAnyExample,
+  SdkArrayExample,
+  SdkArrayType,
+  SdkBodyModelPropertyType,
+  SdkClientType,
+  SdkDictionaryExample,
+  SdkDictionaryType,
+  SdkHttpOperation,
+  SdkHttpOperationExample,
+  SdkHttpParameter,
+  SdkHttpParameterExample,
+  SdkHttpResponse,
+  SdkHttpResponseExample,
+  SdkModelExample,
+  SdkModelPropertyType,
+  SdkModelType,
+  SdkNullExample,
+  SdkServiceMethod,
+  SdkServiceOperation,
+  SdkType,
+  SdkTypeExample,
+  SdkUnionExample,
+  isSdkFloatKind,
+  isSdkIntKind,
+} from "./interfaces.js";
+import { TCGCContext, getValidApiVersion } from "./internal-utils.js";
+import { createDiagnostic } from "./lib.js";
+
+interface LoadedExample {
+  readonly relativePath: string;
+  readonly file: SourceFile;
+  readonly data: any;
+}
+
+/**
+ * Load all examples for a client
+ *
+ * @param context
+ * @param client
+ * @returns a map of all operations' examples, key is operation's operation id,
+ * value is a map of examples, key is example's title, value is example's details
+ */
+async function loadExamples<TServiceOperation extends SdkServiceOperation>(
+  context: TCGCContext,
+  client: SdkClientType<TServiceOperation>
+): Promise<[Map<string, Record<string, LoadedExample>>, readonly Diagnostic[]]> {
+  const diagnostics = createDiagnosticCollector();
+  if (!context.examplesDirectory) {
+    return diagnostics.wrap(new Map());
+  }
+
+  const apiVersion = getValidApiVersion(context, client.apiVersions);
+
+  const exampleDir = apiVersion
+    ? resolvePath(context.examplesDirectory, apiVersion)
+    : resolvePath(context.examplesDirectory);
```

Comment:
This may fail in SDK repo.

Java did a fallback to this
```ts
        exampleDir = version
          ? resolvePath(program.projectRoot, examplesDirectory, version)
          : resolvePath(program.projectRoot, examplesDirectory);
```

Reply:
i suppose the config should be `"{project-root}/examples"`?

Reply:
Yeah, `"{project-root}/examples"` would also do.

But from what I saw on tspconfig, when writing this to typespec-autorest, service would just write "examples".

Reply:
will java fail to get example for `"{project-root}/examples"` according to your code snippet?

Reply:
Yes, we do fallback to the that equivalent to `{project-root}/{whatever-dev-write-ion-exampleDirectory}`.

I am also fine that we require dev to write `"{project-root}/examples"`